R/distribution_formulas.R

```diff
@@ -0,0 +1,370 @@
+#' Validaity checks for new_data_distribution and new_process_distribution
+#' 
+check_distribution_validity <- function(args){
+  family <- args$family
+  families <- args$families
+  sd <- args$sd
+  if (class(family) !=  "family") {
+    cli::cli_abort(c(
+      "{.code {family}} is the incorrect type",
+      "*" = "family needs to be specified as a family class",
+      "*" = " e.g. `family = gaussian()`."))
+  }
+  if (!(family[["family"]] %in% families)) {
+    fam_name <- family[["family"]]
+    cli::cli_abort(c(
+      "FIMS currently does not offer the family, {.code {fam_name}}. for this distribution type",
+      "*" = "The families available for this distribution type are:",
+      "*" = "{.code {families}}."))
+
+  }
+  if(!is.null(args$data_type)){
+    data_type <- args$data_type
+    data_type_names <- c("index", "cpue", "agecomp", "lengthcomp")
```

Comment:
Why is "cpue" an option here?

Reply:
I wasn't sure what descriptior to use here... should it be `landings`, or should `index` be used to describe both fleet and survey data?